build_tools/github_actions/runner/gcp/create_templates.sh

```diff
@@ -0,0 +1,130 @@
+STARTUP_SCRIPT_PATH="/tmp/startup_script.${SHORT_REF}.sh"
+GITHUB_RUNNER_SCOPE=iree-org
+GITHUB_RUNNER_VERSION="2.294.0"
+GITHUB_RUNNER_ARCHIVE_DIGEST="a19a09f4eda5716e5d48ba86b6b78fc014880c5619b9dba4a059eaf65e131780"
+GITHUB_TOKEN_PROXY_URL="https://ght-proxy-zbhz5clunq-ue.a.run.app"
+
+declare -a METADATA=(
+  "github-runner-version=${GITHUB_RUNNER_VERSION}"
+  "github-runner-archive-digest=${GITHUB_RUNNER_ARCHIVE_DIGEST}"
+  "github-runner-config-ref=${TEMPLATE_CONFIG_REF}"
+  "github-runner-config-repo=${TEMPLATE_CONFIG_REPO}"
+  "github-runner-scope=${GITHUB_RUNNER_SCOPE}"
+  "github-token-proxy-url=${GITHUB_TOKEN_PROXY_URL}"
+)
+
+if (( TESTING==1 )); then
+  METADATA+=(github-runner-environment=testing)
+else
+  METADATA+=(github-runner-environment=prod)
+fi
+
+declare -a common_args=(
+  --project=iree-oss
+  # `address=''` indicates an ephemeral IP. This *shouldn't* be necessary here,
+  # as the gcloud docs say that this is the default, but in fact if you leave it
+  # off the VM gets no external IP and is impossible to SSH into. This knowledge
+  # was hard won.
+  --network-interface=network=default,address='',network-tier=PREMIUM
+  --provisioning-model=STANDARD
+  --scopes=https://www.googleapis.com/auth/cloud-platform
+  --no-shielded-secure-boot
+  --shielded-vtpm
+  --shielded-integrity-monitoring
+  --reservation-affinity=any
+  --metadata-from-file=startup-script="${SCRIPT_DIR}/startup_script.sh"
+)
+
+
+function create_template() {
+  local group="$1"
+  local type="$2"
+  if [[ "${group}" == presubmit ]]; then
+    local trust=minimal
+  elif [[ "${group}" == postsubmit ]]; then
+    local trust=basic
+  else
+    echo "Got unrecognized group '${group}'" >2
+    exit 1
+  fi
+
+  local -a metadata=(
+    "${METADATA[@]}"
+    "github-runner-group=${group}"
+    "github-runner-trust=${trust}"
+    "github-runner-labels=${type}"
+  )
+
+  # Join on commas
+  local metadata_string="$(IFS="," ; echo "${metadata[*]}")"
+
+  local -a args=(
+    "${TEMPLATE_BASE_NAME}-${group}-${type}-${SHORT_REF}-${TIME_STRING}"
+    "${common_args[@]}"
+    --service-account="github-runner-${trust}-trust@iree-oss.iam.gserviceaccount.com"
+    --metadata="${metadata_string}"
+  )
+
+  local disk_name="${TEMPLATE_BASE_NAME}-${group}-${type}-${SHORT_REF}-${TIME_STRING}"
+
+  if [[ "${type}" == gpu ]]; then
+    args+=(
+      --machine-type=a2-highgpu-1g
+      --maintenance-policy=TERMINATE
+      --accelerator=count=1,type=nvidia-tesla-a100
+      --create-disk="auto-delete=yes,boot=yes,device-name=${disk_name},image=projects/iree-oss/global/images/github-runner-gpu-2022-08-15-1660603500,mode=rw,size=1000,type=pd-balanced"
+    )
+  elif [[ "${type}" == cpu ]]; then
+    args+=(
+      --machine-type=n1-standard-96
+      --maintenance-policy=MIGRATE
+      --create-disk="auto-delete=yes,boot=yes,device-name=${disk_name},image=projects/iree-oss/global/images/github-runner-2022-07-28-1659048799,mode=rw,size=1000,type=pd-balanced"
```

Comment:
if we add Windows runners, would they be other "types" in this list?

Reply:
I think OS is a different dimension. This script exists mostly to automate the tasks I've been doing by hand in the UI (and routinely messing up). When we get to Windows, we'll have to generalize to more dimensions and then we really need names arguments and such, so I'll probably move to Python.